build/patches/Timezone-Override.patch

```diff
@@ -0,0 +1,1386 @@
+       user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
+ #endif  // !defined(OS_ANDROID)
+ #endif  // !defined(OS_IOS)
++
++  // Bromite #331: Custom Timezone
++  registry->RegisterStringPref(prefs::kContentSettingsCustomTimezone, std::string());
+ }
+ 
+ PrefProvider::PrefProvider(PrefService* prefs,
+@@ -148,6 +151,10 @@ PrefProvider::PrefProvider(PrefService* prefs,
+                             num_exceptions);
+   }
+ 
++  custom_timezone_ = 
++    prefs_->GetString(
++          prefs::kContentSettingsCustomTimezone);
++
+   TRACE_EVENT_END1("startup", "PrefProvider::PrefProvider",
+                    "NumberOfExceptions", num_exceptions);
+ }
+@@ -287,4 +294,14 @@ void PrefProvider::SetClockForTesting(base::Clock* clock) {
+   clock_ = clock;
+ }
+ 
++void PrefProvider::GetPrefTimezoneOverrideValue(std::string& timezone) const {
++  timezone = custom_timezone_;
++}
++
++void PrefProvider::SetPrefTimezoneOverrideValue(const std::string& timezone) {
++  prefs_->SetString(
++        prefs::kContentSettingsCustomTimezone, timezone);
++  custom_timezone_ = timezone;
++}
++
+ }  // namespace content_settings
+diff --git a/components/content_settings/core/browser/content_settings_pref_provider.h b/components/content_settings/core/browser/content_settings_pref_provider.h
+--- a/components/content_settings/core/browser/content_settings_pref_provider.h
++++ b/components/content_settings/core/browser/content_settings_pref_provider.h
+@@ -66,6 +66,9 @@ class PrefProvider : public UserModifiableProvider {
+ 
+   ContentSettingsPref* GetPref(ContentSettingsType type) const;
+ 
++  void GetPrefTimezoneOverrideValue(std::string& timezone) const;
++  void SetPrefTimezoneOverrideValue(const std::string& timezone);
++
+  private:
+   friend class DeadlockCheckerObserver;  // For testing.
+ 
+@@ -103,6 +106,8 @@ class PrefProvider : public UserModifiableProvider {
+ 
+   base::Clock* clock_;
+ 
++  std::string custom_timezone_;
++
+   DISALLOW_COPY_AND_ASSIGN(PrefProvider);
+ };
+ 
+diff --git a/components/content_settings/core/browser/content_settings_registry.cc b/components/content_settings/core/browser/content_settings_registry.cc
+--- a/components/content_settings/core/browser/content_settings_registry.cc
++++ b/components/content_settings/core/browser/content_settings_registry.cc
+@@ -590,6 +590,19 @@ void ContentSettingsRegistry::Init() {
+            ContentSettingsInfo::INHERIT_IF_LESS_PERMISSIVE,
+            ContentSettingsInfo::PERSISTENT,
+            ContentSettingsInfo::EXCEPTIONS_ON_SECURE_AND_INSECURE_ORIGINS);
++
++  // bromite #331: Customize Timezone
++  Register(ContentSettingsType::TIMEZONE_OVERRIDE, "timezone-override", CONTENT_SETTING_ALLOW,
++           WebsiteSettingsInfo::SYNCABLE,
++           WhitelistedSchemes(kChromeUIScheme, kChromeDevToolsScheme),
++           ValidSettings(CONTENT_SETTING_ALLOW,   // use system time 
++                         CONTENT_SETTING_ASK,     // utc
```

Comment:
@uazo is this option corresponding to the custom timezone?

Reply:
yes, "utc" it's wrong, you are right